src/components/EditDeliveryDetailsPopup/EditDeliveryDetailsPopup.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { resetDeliveryDetailsState } from 'redux/deliveryDetailsSlice';
+import { hidePopup } from 'redux/popupSlice';
+import Button from 'components/Button';
+import InnerPopupWrapper from 'components/InnerPopupWrapper';
+import RecipientForm from 'components/DeliveryDetails/RecipientForm';
+import { ReactComponent as CheckmarkIcon } from 'assets/images/greenCheckmark.svg';
+import {
+  ButtonsStyled,
+  ContentStyled,
+  ThankYouPageHeaderStyled,
+  ThankYouPageInfoTextStyled,
+  ThankYouPageStyled
+} from './EditDeliveryDetailsPopupStyled';
+
+const EditDeliveryDetailsPopup = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const [currentStep, setCurrentStep] = useState(1);
+
+  useEffect(() => {
+    return () => {
+      dispatch(resetDeliveryDetailsState());
+    };
+  }, []);
+
+  // get offerTitle from store or props
+  const offerTitle = '1 Month Subscription - Premium Plan';
+
+  return (
+    <InnerPopupWrapper
+      steps={2}
+      isError={false}
+      currentStep={currentStep}
+      popupTitle={t(
+        'edit-delivery-details-popup.title',
+        'Edit delivery details'
+      )}
+    >
+      <ContentStyled>
+        {currentStep === 1 ? (
+          <>
+            <RecipientForm isMyAccount />
+            <ButtonsStyled>
+              <Button theme="simple" onClickFn={() => dispatch(hidePopup())}>
+                {t('edit-delivery-details-popup.button.cancel', 'Back')}
+              </Button>
+              <Button theme="confirm" onClickFn={() => setCurrentStep(2)}>
+                {t(
+                  'edit-delivery-details-popup.button.confirm',
+                  'Update details'
+                )}
+              </Button>
+            </ButtonsStyled>
+          </>
+        ) : (
+          <>
+            <ThankYouPageStyled>
+              <CheckmarkIcon />
+              <ThankYouPageHeaderStyled>
+                {t(
+                  'edit-delivery-details-popup.thank-you-page.header',
+                  'Delivery Details Updated'
+                )}
+              </ThankYouPageHeaderStyled>
+              <ThankYouPageInfoTextStyled>
+                {t(
+                  'edit-delivery-details-popup.thank-you-page.info-text-1',
+                  'Thank you for updating your delivery details for your'
+                )}
+                <p>{offerTitle}</p>
```

Comment:
also let's add a translation key to offer title here, key: `offer-title-${offerId}`